src/main/java/com/microsoft/sqlserver/jdbc/SQLServerPreparedStatement.java

```diff
@@ -1964,6 +1964,16 @@ public final void clearBatch() throws SQLServerException {
                 ArrayList<String> columnList = parseUserSQLForColumnListDW();
                 ArrayList<String> valueList = parseUserSQLForValueListDW(false);
 
+                for (int i = 0; i < valueList.size(); i++) {
+                    if (!valueList.get(i).equals("?")) {
+                        // throw IllegalArgumentException and fallback to original logic for batch insert
+                        throw new IllegalArgumentException(
+                                "Only fully parameterized queries are allowed for using Bulk Copy API for batch insert at the moment.");
+                    }
+                }
+
+                checkAdditionalQuery();
+
                 String destinationTableName = tableName;
                 SQLServerStatement stmt = (SQLServerStatement) connection.createStatement(ResultSet.TYPE_FORWARD_ONLY,
```

Comment:
Statement is created here, and closed in next try's finally block?

Reply:
that try block is closing the resultset. this statement is closed during finally.

Reply:
@peterbae , you should move statement creation to try-with resources.

Reply:
@peterbae , Why don't you just move statement creation to try-with-resources?

Reply:
moved.